### ISSUE TITLE
[linter] Rule stylus/at-rule-empty-line-before does not work well with @extend

### ISSUE THREAD

**Issue by Cactusbone** [stylus/stylelint-stylus#21]
Given the following code

```
$variable
    align-items center

.error-content
    color: #123456
    @extend $variable

    &__left
        text-align left
```

gives a `Expected empty line before at-rule  stylus/at-rule-empty-line-before` error, and --fix adds a line before @extend

Moreover

```
$variable
    align-items center

.error-content
    color: #123456

    @extend $variable
    &__left
        text-align left
```

does not raise error despite `&__left` not having an empty line on the line before

**Comment by ota-meshi** [comment]
I didn't understand what you thought was the problem.
`stylus/at-rule-empty-line-before` that  require an empty line before at-rules. (You can disallow it by config.)
https://stylus.github.io/stylelint-stylus/rules/at-rule-empty-line-before.html
Your example seemed to work fine to me.

**Comment by Cactusbone** [comment]
sorry about the delay, I'll have to look at notifications configuration :/ 

`@extend $variable` is not a rule, so I wasn't expecting to get a line before it.

However `&__left` is one, so not adding it after the ` @extend $variable` seems weird to me
